app/src/screens/ActiveCall/hooks.ts

```diff
@@ -114,67 +156,41 @@ const useHangup = (
  * @returns - Handlers and state for the audio device selection button.
  */
 const useSelectAudioOutputDevice = (
-  activeCall: ActiveCall,
+  activeCall: ActiveCall | undefined,
   audioDevices: AudioDevicesState,
   dispatch: ReturnType<typeof useTypedDispatch>,
 ) => {
-  const [handle, isDisabled, isActive] = React.useMemo(() => {
-    if (activeCall?.status !== 'fulfilled') {
-      return [undefined, true, false];
-    }
-
-    if (activeCall.callInfo.state === 'disconnected') {
-      return [undefined, true, false];
-    }
-
-    if (audioDevices?.status !== 'fulfilled') {
-      return [undefined, true, false];
-    }
-
-    const earpieceDevice = audioDevices.audioDevices.find(
-      (dev) => dev.type === TwilioAudioDevice.Type.Earpiece,
-    );
-
-    const speakerDevice = audioDevices.audioDevices.find(
-      (dev) => dev.type === TwilioAudioDevice.Type.Speaker,
-    );
-
-    if (
-      typeof earpieceDevice === 'undefined' ||
-      typeof speakerDevice === 'undefined'
-    ) {
-      return [undefined, true, false];
-    }
-
-    if (audioDevices.selectedDevice === null) {
-      return [
-        () =>
-          dispatch(selectAudioDevice({ audioDeviceUuid: earpieceDevice.uuid })),
-        false,
-        false,
-      ];
-    }
+  const [handle, isDisabled, isActive] = React.useMemo(
+    () =>
+      match<
+        [typeof activeCall, typeof audioDevices],
+        [() => void, boolean, boolean]
+      >([activeCall, audioDevices])
+        .with(
+          [{ info: { state: 'connected' } }, { status: 'fulfilled' }],
+          ([_, au]) =>
```

Comment:
@kpchoy I agree with you 100%. Using `ts-pattern` definitely feels more obscure for the average TS developer, and in some of my use-cases I think it's more complicated than it needs to be. This is a prime example, since I have to rename variables for them to have the proper typing. In this case, `au` is simply `audioDevices` but with the right typings. Ideally, I wouldn't have to rename things but that's the way the library works, limitation of TS since it doesn't have true pattern matching baked into the language. In other functional languages, the standard is to postfix the variable with a `'` denoting prime, so it would be `audioDevices'` or "audioDevices prime". I was thinking we could use `_` instead? WDYT?